_episodes/03-monitor-the-model.md

```diff
@@ -0,0 +1,591 @@
+> > We can then compare those to the true labels, for instance by
+> > ~~~
+> > fig, axes = plt.subplots(1, 2, figsize=(12, 6))
+> > plt.style.use('ggplot')  # optional, that's only to define a visual style
+> > axes[0].scatter(y_train_predicted, y_train, s=10, alpha=0.5, color="teal")
+> > axes[0].set_title("training set")
+> > axes[0].set_xlabel("predicted sunshine hours")
+> > axes[0].set_ylabel("true sunshine hours")
+> > 
+> > axes[1].scatter(y_test_predicted, y_test, s=10, alpha=0.5, color="teal")
+> > axes[1].set_title("test set")
+> > axes[1].set_xlabel("predicted sunshine hours")
+> > axes[1].set_ylabel("true sunshine hours")
+> > ~~~
+> > {: .language-python}
+> > ![Scatter plot to evaluate training and test set](../fig/03_regression_training_test_comparison.png)
+> > Maybe that is not exactly what you expected? What is the issue here? Any ideas?
+> > 
+> > The accuracy on the training set is fairly good. 
+> > In fact, considering that the task of predicting the daily sunshine hours is really not easy it might even be surprising how well the model predicts that 
+> > (at least on the training set). Maybe a little too good?
+> > For those familiar with (classical) machine learning this might look familiar. 
+> > It is a very clear signature of **overfitting** which means that the model has to some extend memorized aspects of the training data. 
+> > As a result makes much more accurate predictions on the training data than on unseen data.
+> {:.solution}
+{:.challenge}
+
+Overfitting also happens in classical machine learning, but there it is usually interpreted as the model having more parameters than the training data would justify (say, a decision tree with too many branches for the number of training instances). As a consequence one would reduce the number of parameters to avoid overfitting.
+In deep learning the situation is slightly different. It can -same as for classical machine learning- also be a sign of having a *too big* model, meaning a model with too many parameters (layers and/or nodes). However, in deep learning higher number of model parameters are often still considered acceptable and models often perform best (in terms of prediction accuracy) when they are at the verge of overfitting. So, in a way, training deep learning models is always a bit like playing with fire...
+
+## Watch your model training closely
+As we just saw, deep learning models are prone to overfitting. Instead of iterating through countless cycles of model trainings and subsequent evaluations with a reserved test set, it is common practice to work with a 2nd split off dataset to monitor the model during training. This is the *validation set* which can be regarded as a 2nd test set. As with the test set the datapoints of the *validation set* are not used for the actual model training itself. Instead we evalute the model with the *validation set* after every epoch during training, for instance to splot if we see signs of clear overfitting.
```

Comment:
```suggestion
As we just saw, deep learning models are prone to overfitting. Instead of iterating through countless cycles of model trainings and subsequent evaluations with a reserved test set, it is common practice to work with a second split off dataset to monitor the model during training. This is the *validation set* which can be regarded as a second test set. As with the test set the datapoints of the *validation set* are not used for the actual model training itself. Instead we evaluate the model with the *validation set* after every epoch during training, for instance to splot if we see signs of clear overfitting.
```